.github/workflows/rust.yml

```diff
@@ -26,15 +26,142 @@ jobs:
       - run: rustup show
 
       - uses: Swatinem/rust-cache@v2
-      - run: cargo fmt -- --check
-      - run: cargo doc --no-deps --document-private-items
+      - run: cargo fmt -p relay -- --check
+      - run: cargo doc -p relay --no-deps --document-private-items
         env:
           RUSTDOCFLAGS: "-D warnings"
-      - run: cargo clippy -- -D warnings
-      - run: cargo test
+      - run: cargo clippy -p relay -- -D warnings
+      - run: cargo test -p relay
 
-  cross: # cross is separate from test because cross-compiling yields different artifacts and we cannot reuse the cache.
-    name: Cross compile all crates
+  test-connlib:
+    needs: draft-release
+    name: Static analysis
+    strategy:
+      matrix:
+        rust: [stable, nightly]
+        runs-on:
+          - ubuntu-20.04
+          - ubuntu-22.04
+          - macos-11
+          - macos-12
+          - windows-2019
+          - windows-2022
+        # FIXME: There's this weird cargo thing where it stops finding the webrtc's related crates
+        # probably has to do with the cache + some cargo bug. 
+        exclude:
+          - rust: nightly
+            runs-on: ubuntu-20.04
+    runs-on: ${{ matrix.runs-on }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Update toolchain
+        run: rustup update --no-self-update ${{ matrix.rust }} && rustup default ${{ matrix.rust }} && rustup component add clippy
```

Comment:
I added a `rust-toolchain.toml` to the workspace, meaning cargo will automatically install and use that version. Installing it in CI is as easy (although not intuitive) as `rustup show`. See the other CI checks above.